robocorp-code/vscode-client/src/extension.ts

```diff
@@ -324,6 +335,85 @@ async function cloudLogoutAndRefresh() {
     refreshCloudTreeView();
 }
 
+async function convertProject() {
+    const DEFAULT_ERROR_MSG = `
+            Could not convert project.
+            Please check the output logs for more details.
+            `;
+
+    const DEFAULT_ERROR_STATUS = "Error while converting project.";
+
+    try {
+        const converterLocation = await ensureConvertBundle(true);
+        if (!converterLocation) {
+            throw new Error("Converter bundle was not found. Possible download issues.");
+        }
+        const converterBundle = require(converterLocation);
+
+        // let the user decide what type of project will be converted
+        const vendorMap = {
+            "Blue Prism": "blueprism",
+            "UiPath": "uipath",
+            "Automation Anywhere 360": "a360",
+        };
+        const items = Object.keys(vendorMap);
+        let selectedVendor = await window.showQuickPick(items, {
+            "placeHolder": "Please select the format of current project",
```

Comment:
I think the logic should first get the active editor (and validate it if it's valid as the user may not have an active editor) and then show in the title the name of the file that'll be converted (just so that the user know we're converting the active editor).

Also, not sure if it's possible, but if we know that the file extension (say, .xml), we could do some pre-validation and say that it's wrong from the get-go (i.e.: if the user opens a .robot file and asks for the conversion we already know it won't work...).

Reply:
I've changed the logic.
But what should happen if there is no active editor? Should we open a file picker and open the file in the active editor and continue with the conversion? 🤔 

In regards to the file format auto-detection, we don't have an implemented solution yet. 
I did more of a superficial one once, but people were a bit against the idea as we don't know what extensions or formats of the exported 3rd parties projects. But I do think a task for this is in progress (or at least we should issue a reminder). The fail fast idea is still valid, but the format validation should be part of the converter so it becomes a common tool.

Reply:
Agreed, that validation should be a part of the converter.